src/typ.rs

```diff
@@ -2396,6 +2396,26 @@ fn infer_clause_guard(
                 right: Box::new(right),
             })
         }
+        
+        ClauseGuard::GreaterThan {
+            location,
+            left,
+            right,
+            ..
+        } => {
+            let left = infer_clause_guard(*left, level, env)?;
+            unify(int(), left.typ(), env)
+                .map_err(|e| convert_unify_error(e, left.location()))?;
+            let right = infer_clause_guard(*right, level, env)?;
+            unify(int(), right.typ(), env)
+                .map_err(|e| convert_unify_error(e, right.location()))?;
```

Comment:
Ok so here we are just checking the integer types. But I'm guessing rust is able to use the same comparison operator `>` `<` for many different types 🤔 

If that's the case would it be possible to make a broader check? Kind of like using a `Rust::Numeric` type or the like and checking for "numeric" + same type.

Reply:
Gleam doesn't have any features like operator overloading or type classes so the `>` operator only works on Ints. For floats we use `>.`

We might add something here later, but this is what we have for now :)

Reply:
Right, I forget that we're just using Rust to parse valid Gleam.